clients/segments/segment_test.go

```diff
@@ -60,22 +57,33 @@ func TestList(t *testing.T) {
     }
   ]`
 
-	mockClient := segments.NewMockclient(gomock.NewController(t))
-	mockClient.EXPECT().
-		List(gomock.AssignableToTypeOf(t.Context()), gomock.AssignableToTypeOf(rest.RequestOptions{})).
-		Return(&http.Response{
-			StatusCode: http.StatusOK,
-			Body:       io.NopCloser(strings.NewReader(apiResponse)),
-		}, nil)
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
[nit]: You could actually split the request and not handle everything in here.
For example, https://github.com/Dynatrace/dynatrace-configuration-as-code/blob/f0021e8adcb7a6d59b98170385bc69ce3c8b45a8/pkg/client/dtclient/settings_client_test.go#L941-L962

Reply:
Should be improved, totally agree with you. But in my opinion in the next step.